app/scripts/migrations/111.ts

```diff
@@ -0,0 +1,44 @@
+import { cloneDeep } from 'lodash';
+import { hasProperty, isObject } from '@metamask/utils';
+
+type VersionedData = {
+  meta: { version: number };
+  data: Record<string, unknown>;
+};
+
+export const version = 111;
+
+/**
+ * Reset all values for SelectedNetworkController.state.domains
+ * These values are associated with an experimental feature flag and should be reset before proceeding with
+ * the feature development.
+ *
+ * @param originalVersionedData - Versioned MetaMask extension state, exactly what we persist to dist.
+ * @param originalVersionedData.meta - State metadata.
+ * @param originalVersionedData.meta.version - The current state version.
+ * @param originalVersionedData.data - The persisted MetaMask state, keyed by controller.
+ * @returns Updated versioned MetaMask extension state.
+ */
+export async function migrate(
+  originalVersionedData: VersionedData,
+): Promise<VersionedData> {
+  const versionedData = cloneDeep(originalVersionedData);
+  versionedData.meta.version = version;
+  transformState(versionedData.data);
+  return versionedData;
+}
+
+function transformState(state: Record<string, any>) {
+  if (
+    !hasProperty(state, 'SelectedNetworkController') ||
+    !isObject(state.SelectedNetworkController) ||
```

Comment:
Nit: In other migrations we distinguished between the "Missing state that is expected to be missing" scenarios, and "Malformed/corrupted state".

We would expect the `SelectedNetworkController` state to be missing if the user upgraded from an older wallet, so returning `state` unchanged for that scenario seems good. But if that state exists, the remaining three conditions should be true for all users unless the state was somehow corrupted. So we could return an error in those scenarios rather than returning state unchanged, to alert us to that state corruption.

Reply:
added error logging here: https://github.com/MetaMask/metamask-extension/pull/23130/commits/bae4ca69e0077ef03f7c781c066cd31d3698e7e0